lib/tic_tac_toe.rb

```diff
@@ -59,5 +60,95 @@ def board_full?
       !self.include?(:available)
     end
 
+    def best_move
+      if cpu_check_for_wins_or_blocks_horizontally?(:player_one)
+      elsif cpu_check_for_wins_or_blocks_vertically?(:player_one)
+      elsif cpu_check_for_wins_or_blocks_diagonally?(:player_one)
+  
+      elsif cpu_check_for_wins_or_blocks_horizontally?(:player_two)
+      elsif cpu_check_for_wins_or_blocks_vertically?(:player_two)
+      elsif cpu_check_for_wins_or_blocks_diagonally?(:player_two)
+      else
+         cpu_take_open_space()
+      end
+      @best_move
+    end
+
+    #CPU LOGIC METHODS
+    def open_spaces
+      spaces_available_index = self.each_index.select {|i| self[i] == :available}
+      spaces_available_index.map {|each| each}
+    end
+
+    def cpu_take_open_space
+      @best_move = self.open_spaces().sample
+    end
+  
+  def cpu_check_for_wins_or_blocks_horizontally?(player_symbol)
+    if self[0] == player_symbol && self[1] == player_symbol && self[2] == :available
```

Comment:
This seems like a lot of duplication.
Can't we do some sort of for each with `WIN_COMBOS`
```suggestion
WIN_COMBOS.each do |win_combo|
   # if two of the spaces == player and one  of the spaces == :avaliable  return the available space
end

```